### PR TITLE
feat(ui): Limit release files and commits

### PR DESCRIPTION
**Pull request by matejminar**
This PR adds pagination to release `Commits` and limits the number of returned `Files Changed` (we don't want to have pagination there).

#sync-getsentry

### CODE REVIEW

File: src/sentry/static/sentry/app/views/releasesV2/detail/filesChanged/index.tsx [hunk 1]
```diff
@@ -65,14 +82,35 @@ class FilesChanged extends AsyncView<Props, State> {
...
-          Object.keys(filesByRepository).map(repository => (
-            <RepositoryFileSummary
-              key={repository}
-              repository={repository}
-              fileChangeSummary={filesByRepository[repository]}
-              collapsable={false}
-            />
-          ))
+          <React.Fragment>
+            {Object.keys(filesByRepository).map(repository => (
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/releasesV2/detail/filesChanged/index.tsx:86]: nb: you could use `Object.entries()` since you use key + value

**Review by billyvg — APPROVED** [review]
lgtm

File: src/sentry/api/endpoints/filechange.py [hunk 2]
```diff
@@ -31,12 +32,17 @@ def get(self, request, organization, version):
         if not self.has_release_permission(request, organization, release):
             raise ResourceDoesNotExist
 
+        try:
+            limit = int(request.GET.get("limit"))
+        except ValueError:
+            limit = None
```
Comment (markstory) [inline src/sentry/api/endpoints/filechange.py:38]: Could we enforce a limit of sentry.api.paginator.MAX_LIMIT which is 100

File: src/sentry/static/sentry/app/views/releasesV2/list/index.tsx [hunk 3]
```diff
@@ -182,7 +182,7 @@ class ReleasesList extends AsyncView<Props, State> {
       return <LoadingIndicator />;
     }
 
-    if (!releases.length) {
+    if (!releases?.length) {
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/releasesV2/list/index.tsx:185]: :clap:

File: src/sentry/static/sentry/app/views/releasesV2/detail/filesChanged/index.tsx (hunk 1 above)
Comment (matejminar) [reply src/sentry/static/sentry/app/views/releasesV2/detail/filesChanged/index.tsx]: Good idea 👍

File: src/sentry/api/endpoints/filechange.py (hunk 2 above)
Comment (matejminar) [reply src/sentry/api/endpoints/filechange.py:39]: The way it worked in old releases for ages is that this endpoint returned ALL unpaginated results - now we want to show in the frontend first 200 and then there's a button called show all files which will load everything. I would like to revisit this behavior, this is more of a quick fix for one specific edge case.

Comment (markstory) [reply src/sentry/api/endpoints/filechange.py:39]: Sounds good.

**Pull request merged**